### PR TITLE
fix off by one in API

### PR DESCRIPTION
**Pull request by lilisierrayu**
**Patch Description**
The prompt_len in current code relies on the batch_generator implementation and can easily introduce error. Remove that dependence in this PR. 

**Testing steps**
Describe how you tested your changes

### CODE REVIEW

File: metaseq/hub_utils.py [hunk 1]
```diff
@@ -592,7 +592,9 @@ def generate(
 
             logger.info(f"Preparing generator with settings {self.cfg.generation}")
             generator = self.task.build_generator(
-                self.models, self.cfg.generation, extra_gen_cls_kwargs={"stop": stop}
+                self.models,
+                self.cfg.generation,
+                extra_gen_cls_kwargs={"stop": stop, "need_logprobs": True},
```
Comment (stephenroller) [inline metaseq/hub_utils.py:597]: need log probs results in a lot more memory being used. we shouldn't use this unless we absolutely have to

Comment (lilisierrayu) [reply metaseq/hub_utils.py:597]: That's reasonable. How about do: 
```
if logprobs > 0:
    generator.need_logprobs = True
```

Comment (lilisierrayu) [reply metaseq/hub_utils.py]: Change applied.

Comment (stephenroller) [reply metaseq/hub_utils.py]: Perfect

**Review by stephenroller — APPROVED** [review]
Approving conditional on change discussed

**Pull request merged**